src/Scaler/Services/ScalerMetadata.cs

```diff
@@ -46,7 +62,19 @@ private string LeaseAccountHost
 
         public static ScalerMetadata Create(ScaledObjectRef scaledObjectRef)
         {
-            return JsonConvert.DeserializeObject<ScalerMetadata>(scaledObjectRef.ScalerMetadata.ToString());
+            var metadata = JsonConvert.DeserializeObject<ScalerMetadata>(scaledObjectRef.ScalerMetadata.ToString());
+
+            if (string.IsNullOrEmpty(metadata.Connection) && string.IsNullOrEmpty(metadata.ConnectionFromEnv))
```

Comment:
Good selection of exception and exception message. Could replace the if condition with `if (metadata.Connection is null)` since it will be null only when both properties are null in input `scaledObjectRef.ScalerMetadata`.